### DISCUSSION TITLE
Specific Calendar not showing on EXT2

### DISCUSSION THREAD

**Discussion by lefebe** [eouia/MMM-CalendarExt2#307] [Q&A] [answered]
I have a total of 5 different calendars set to show on my EXT2, but one calendar REFUSES to show.  It was there for a while then it disappeared.  I've tried switching the URL with another of the existing/working calendars= nothing.  I've tried importing the events into a new calendar and using that link with no results.  I've tried removing all other calendars EXCEPT this one with no results.

I can't imagine that it would be because there are too many events (1-2 daily).
*All of my calendars are shared apple calendars*

Has anyone run into this issue before?

**Comment by beeperr** [comment]
I had the same thing happen to me recently and am also looking for help. I run four google calendars on my EXT2 and have for several years now. This summer, one of the calendars stopped updating—i.e., when I would add new events, they wouldn't show up. I'm running the same MM on two different screens. To troubleshoot, I updated MM and then the MMM-CalendarExt2 module (neither had been updated in over a year). The MM update broke most of my modules (whoops), but Ext2 continues work for only three of my four calendars. 

Looking into the pm2 logs, I found the following error right after it started scanning the calendar:

0|mm     | [08.09.2023 16:49.11.603] [LOG]   [CALEXT2] calendar:BPR Work >> Scanning start with interval:300000
.
.
.
0|mm     | [08.09.2023 16:49.26.472] [LOG]   [CALEXT2] calendar:BPR Work >> Cannot read properties of null (reading 'replace')
0|mm     | [08.09.2023 16:49.27.932] [ERROR] TypeError: Cannot read properties of null (reading 'replace')
0|mm     |     at /home/pi/MagicMirror/modules/MMM-CalendarExt2/node_helper.js:170:31
0|mm     |     at Array.forEach (<anonymous>)
0|mm     |     at Class.parser (/home/pi/MagicMirror/modules/MMM-CalendarExt2/node_helper.js:149:17)
0|mm     |     at /home/pi/MagicMirror/modules/MMM-CalendarExt2/node_helper.js:41:16
0|mm     |     at Class.scanCalendar (/home/pi/MagicMirror/modules/MMM-CalendarExt2/node_helper.js:102:7)
0|mm     |     at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
0|mm     | [08.09.2023 16:49.27.956] [ERROR] (node:2243) UnhandledPromiseRejectionWarning: TypeError: Cannot read properties of undefined (reading 'text')
0|mm     |     at Class.scanCalendar (/home/pi/MagicMirror/modules/MMM-CalendarExt2/node_helper.js:110:46)
0|mm     |     at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
0|mm     | (Use `electron --trace-warnings ...` to show where the warning was created)
0|mm     | [08.09.2023 16:49.27.957] [ERROR] (node:2243) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 1)

My best guess it that I have some calendar event that it can't read so it throws it out of whack. ¯\_(ツ)_/¯

**Comment by beeperr** [comment, marked as answer]
Responding to my own issue 8 minute later...

I was able to resolve my issue. I went and looked at a list of my calendar events, and there was one all-day event with no title—probably an accidental creation bc Apple's calendar app sucks. I deleted the event, restarted MM, and all is working now. 

Good day.
